### PR TITLE
Feature/installation percentage

### PR DESCRIPTION
**Pull request by harroyoc**
- [ ] Add a link to the ticket in the PR title or add a description of work in the PR title
- [ ] dashcam firmware was loaded on a device and successfully runs
- [ ] semantic version was incremented inside `src/config/index.ts`

### CODE REVIEW

**Comment by HariHive** [comment]
We are still unaware of progress of Unzip command right ?

File: src/util/firmware.ts [hunk 1]
```diff
@@ -44,27 +47,53 @@ class FirmwareManager {
...
   private runSpawn(cmd: string) {
     this.updateStatus(InstallStatus.Installing, 'Installation in progress...');
 
     const child = spawn(cmd, { shell: true });
+    let output = '';
 
     child.stdout.setEncoding('utf8');
     child.stdout.on('data', data => {
-      this.updateStatus(InstallStatus.Installing, data.toString());
+      output += data.toString();
```
Comment (HariHive) [inline src/util/firmware.ts:89]: Please correct me if I am missing something here:
we can also do: output = data.toString() right ?
as every line has percentage value
this way we do not have to loop through whole output and just parse last line

Comment (HariHive) [reply src/util/firmware.ts:89]: if we are concerned that not every line has percentage, then we can probably store it in a var like "lastKnownPercent" value ?

Comment (harroyoc) [reply src/util/firmware.ts:89]: `output = data.toString()`
No , that would output only the streaming portion received like "...."

Comment (harroyoc) [reply src/util/firmware.ts:89]: `We are still unaware of progress of Unzip command right ?`
Im ok with that, it's just a few seconds

Comment (HariHive) [reply src/util/firmware.ts:89]: > if we are concerned that not every line has percentage, then we can probably store it in a var like "lastKnownPercent" value ?
so this is still a possibility right ?

Comment (harroyoc) [reply src/util/firmware.ts:89]: not sure I understand this:
`if we are concerned that not every line has percentage, then we can probably store it in a var like "lastKnownPercent" value ?`

Comment (harroyoc) [reply src/util/firmware.ts:89]: I prefer not to store any value because we would need to "clean it"

Comment (harroyoc) [reply src/util/firmware.ts:89]: so the regex will give me the last percentage found... so if we are on 27%... and for the next 10 lines there is no percentage we still send 27

File: src/routes/firmware.ts [hunk 2]
```diff
@@ -20,4 +22,9 @@ router.get('/progress', async (req, res) => {
   res.json(firmwareManager.getProgress());
 });
 
+router.post('/complete', async (req, res) => {
+  exec('reboot'); // Restart the dashcam to apply the new firmware
+  res.json({});
```
Comment (punov) [inline src/routes/firmware.ts:27]: It's funny it will most probably not even get to this line, be ready for that!

**Review by punov — APPROVED** [review]

File: src/routes/firmware.ts (hunk 2 above)
Comment (harroyoc) [reply src/routes/firmware.ts:27]: true

**Pull request merged**